### ISSUE TITLE
Passwordless auth does not allow SSL

### ISSUE THREAD

**Issue by ahachete** [neondatabase/neon#3416]
## Steps to reproduce

```
$ psql 'postgres://pg.neon.tech?sslmode=require'
psql: error: connection to server at "pg.neon.tech" (54.200.201.103), port 5432 failed: server does not support SSL, but SSL was required
```

## Expected result

Connection should succeed.

## Description

SSL connections should be a must for publicly hosted servers. Users may be unaware of private or sensitive data being exposed via unencrypted connections when connecting via the passwordless authentication mechanism.

While I appreciate the innovation and convenience of this authentication mechanism, IMHO security should come first. If I'm not mistaken, I imagine adding SSL support is not possible --precisely SSL prevents middleware/proxies.

This is not applicable to other authentication mechanisms (with username, password and connection string), where SSL is enabled.

**Comment by kelvich** [comment]
Yes, thanks for reporting! That's that one of the checklist items to fix before going to GA. Actually TLS is supported, so we need to change a configuration.

> If I'm not mistaken, I imagine adding SSL support is not possible --precisely SSL prevents middleware/proxies.

Proxy parses postgres protocol so acts as an endpoint here. Hence it is possible and there is no man-in-the-middle for TLS connection.

**Comment by ahachete** [comment]
> Proxy parses postgres protocol so acts as an endpoint here.

Oh, if proxy terminates SSL and is not end-to-end TLS, yes.
